src/Formats/FormatFactory.h

```diff
@@ -127,6 +127,10 @@ class FormatFactory final : private boost::noncopyable
     /// The checker should return true if format support append.
     using SubsetOfColumnsSupportChecker = std::function<bool(const FormatSettings & settings)>;
 
+    /// Some formats like ORC can support reading subset of subcolumns depending on settings.
+    /// The checker should return true if format support append.
```

Comment:
I think there is no settings that can disable it for ORC/Parquet. We can use simple bool flag instead of checker function

Reply:
We need checker function because some formats has serveral implementations. Checker function makes it flexible. 

For example, format orc has two implementations: `NativeORCBlockInputFormat` and `ORCBlockInputFormat`, choosing which one is controlled by format_settings. Currently `NativeORCBlockInputFormat` does support tuple field pruning and `ORCBlockInputFormat` does not, which means marking supports_subset_of_subcolumns  of `ORCBlockInputFormat` as true doesn't speedup performance nor slowdown.